Source/WebCore/dom/Document.h

```diff
@@ -2269,7 +2278,7 @@ class Document
 
     bool m_updateTitleTaskScheduled { false };
 
-    FocusTrigger m_latestFocusTrigger { FocusTrigger::Other };
+    FocusTrigger m_latestFocusTrigger;
```

Comment:
I mean, you could, but would that be a good idea? The order of the enumeration values may change in the future, and it's not obvious that this call site would need to change as well. Also, zero may not be a valid enumeration value, and, so this could possibly lead to initializing m_latestFocusTrigger with an invalid value. I just checked and there is no error or warning thrown in this case.

Reply:
I do think it’s a good idea.

Generally speaking, uninitialized memory is even worse than the things you mentioned. Can lead to unpredictable behavior and even security vulnerabilities. Reliably initializing to a value that we can then overwrite in the constructor if it’s not really a suitable default is mild by comparison.

Typical enumerations will have a zero-valued item, typically the first one, and in future we can try to make sure that first value is a good default for most situations, just as we do with a class when we decide what the default constructor does.

Reply:
I see, you're interested in protecting against a programming error where someone fails to correctly initialize the instance variable in the constructor. I'd prefer that we'd protect against this with the compiler, but it looks like clang's `-Wuninitialized` doesn't catch this case (but GCC does), so maybe we can't rely on the compiler here.

Okay, so long as we re-initialize to a explicit value in the implementation file, it seems reasonable to zero-init in the header first.

Reply:
I agree with Darin: the `{ }` initialization in the class body is worth it. Uninitialized memory should be avoided at all costs, even if we expect it to be reinitialized later.